src/path/Path.js

```diff
@@ -404,8 +404,10 @@ var Path = PathItem.extend(/** @lends Path# */{
                 this._updateSelection(segment, 0, segment._selection);
         }
         if (append) {
-            // Append them all at the end by using push
-            segments.push.apply(segments, segs);
+            // Append them all at the end.
+            // Use a loop as it is the best way to handle big arrays (see #1493)
+            for (var i = 0; i < segs.length; i++)
+                segments.push(segs[i]);
```

Comment:
Some code-style remarks:

- Let's use blocks after `if` and `for` (unless the statement is super simple). I know the code-base doesn't do this currently in some places, but I tend to disagree with this by now. We shouldn't minify code by hand :) 

- Let's use this loop optimisation: `for (var i = 0, l = segs.length; i < l; i++)`

Result:

```js
for (var i = 0, l = segs.length; i < l; i++) {
    segments.push(segs[i]);
}
```

Reply:
- Sorry again for the style, I put that in my IDE rules too :)
- OK for the loop optimisation